### DISCUSSION TITLE
Is there a way to make levels procedural by code?

### DISCUSSION THREAD

**Discussion by ares-rxu** [blackears/cyclopsLevelBuilder#163] [Q&A] [answered]
Something like mesh library maybe, I want to make a dungeon or more stuff and just looked into Gaea and thought if it can work together. I would use GDScript

**Comment by blackears** [comment, marked as answer]
You can use a script to procedurally generate CyclopsBlock objects, or any other Godot mesh objects too.  There is no dungeon generation feature built into Cyclops at the moment, but that might be an interesting idea for future development.

**Reply by ares-rxu** [reply]
Can you reference the doc or class for this? An example would be great, also I already made the dungeon generator work around.
I made scenes with cyclops blocks as tiles, then used walker 2d of Gaea on a tile map and put on ready function a for loop on all tiles and get tiles from a tile set representing the scenes I made. Then just spawn all tiles from scenes into 3d

**Reply by blackears** [reply]
There's no document describing it so you'll need to look at the source code.  The ConvexBlock itself is pretty simple - the trick part is setting the ConvexVolumeData which may not be so straight forward.  Basically it's a set of arrays, each of which contain a property of a vertex, edge or face.  If you've ever used OpenGL before, it uses a similar way to organizing data.  You can also take a look at ConvexVolume which has a method for turning the data object into a mesh that can be used more intuitively.

I'm also planning to some major work on this soon, so this may be changing soon.

**Comment by JekSun97** [comment]
it would be interesting to see such an opportunity
